test/src/test/java/jenkins/tasks/SimpleBuildStepTest.java

```diff
@@ -0,0 +1,74 @@
+package jenkins.tasks;
+
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.Action;
+import hudson.model.Build;
+import hudson.model.BuildListener;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Label;
+import hudson.model.ParametersAction;
+import hudson.model.Project;
+import hudson.model.Run;
+import hudson.model.Slave;
+import hudson.model.StringParameterValue;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStep;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import java.io.IOException;
+import javax.annotation.Nonnull;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.TestExtension;
+
+public class SimpleBuildStepTest {
+
+    private static class StepThatGetsEnvironmentContents extends Builder implements SimpleBuildStep {
+
+        @Override
+        public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull EnvVars env, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws InterruptedException, IOException {
+            // Check that the environment we get includes values from the slave
+            Assert.assertEquals("JENKINS-29144", env.get("TICKET"));
+            // FIXME: Should this test any other envvars? Or Parameters?
```

Comment:
```suggestion
```

I suggested `ParametersAction` as a somewhat simpler test setup, but I do not suppose it really matters.